util/linuxfw/iptables.go

```diff
@@ -0,0 +1,821 @@
+				return err
+			}
+		}
+
+		entsBuf = entsBuf[parser.offset:]
+		totalOffset += parser.offset
+	}
+	return nil
+}
+
+// TODO(andrew): convert to use cstruct
+type entryParser struct {
+	buf    []byte
+	offset int
+
+	logf logger.Logf
+
+	// Set to 'true' to print debug messages about unused bytes returned
+	// from the kernel
+	checkExtraBytes bool
+}
+
+func (p *entryParser) haveLen(ln int) bool {
+	if len(p.buf)-p.offset < ln {
+		return false
+	}
+	return true
+}
+
+func (p *entryParser) assertLen(ln int) error {
+	if !p.haveLen(ln) {
+		return fmt.Errorf("need %d bytes: %w", ln, errBufferTooSmall)
+	}
+	return nil
+}
+
+func (p *entryParser) getBytes(amt int) []byte {
+	ret := p.buf[p.offset : p.offset+amt]
+	p.offset += amt
+	return ret
+}
+
+func (p *entryParser) getByte() byte {
+	ret := p.buf[p.offset]
+	p.offset += 1
+	return ret
+}
+
+func (p *entryParser) get4() (ret [4]byte) {
+	ret[0] = p.buf[p.offset+0]
+	ret[1] = p.buf[p.offset+1]
+	ret[2] = p.buf[p.offset+2]
+	ret[3] = p.buf[p.offset+3]
+	p.offset += 4
+	return
+}
+
+func (p *entryParser) setOffset(off, max int) error {
+	// We can't go back
+	if off < p.offset {
+		return fmt.Errorf("invalid target offset (%d < %d): %w", off, p.offset, errMalformed)
+	}
+
+	// Ensure we don't go beyond our maximum, if given
+	if max >= 0 && off >= max {
+		return fmt.Errorf("invalid target offset (%d >= %d): %w", off, max, errMalformed)
+	}
+
+	// If we aren't already at this offset, move forward
+	if p.offset < off {
+		if p.checkExtraBytes {
+			extraData := p.buf[p.offset:off]
+			diff := off - p.offset
+			p.logf("%d bytes (%d, %d) are unused: %s", diff, p.offset, off, hex.EncodeToString(extraData))
+		}
+
+		p.offset = off
+	}
+	return nil
+}
+
+var (
+	errBufferTooSmall = errors.New("buffer too small")
+	errMalformed      = errors.New("data malformed")
+)
+
+type entry struct {
+	Offset      int
+	IP          iptip
+	NFCache     uint32
+	PacketCount uint64
+	ByteCount   uint64
+	Matches     []match
+	Target      target
+}
+
+func (e entry) String() string {
+	var sb strings.Builder
+	sb.WriteString(`{`)
+
+	fmt.Fprintf(&sb, `Offset:%d IP:%v PacketCount:%d ByteCount:%d`, e.Offset, e.IP, e.PacketCount, e.ByteCount)
+	if len(e.Matches) > 0 {
+		fmt.Fprintf(&sb, ` Matches:%v`, e.Matches)
+	}
+	fmt.Fprintf(&sb, ` Target:%v`, e.Target)
+
+	sb.WriteString(`}`)
+	return sb.String()
+}
+
+func (p *entryParser) parseEntry(b []byte) (entry, error) {
+	startOff := p.offset
+
+	iptip, err := p.parseIPTIP()
+	if err != nil {
+		return entry{}, fmt.Errorf("parsing IPTIP: %w", err)
+	}
+
+	ret := entry{
+		Offset: startOff,
+		IP:     iptip,
+	}
+
+	// Must have space for the rest of the members
+	if err := p.assertLen(28); err != nil {
+		return entry{}, err
+	}
+
+	ret.NFCache = native.Endian.Uint32(p.getBytes(4))
+	targetOffset := int(native.Endian.Uint16(p.getBytes(2)))
+	nextOffset := int(native.Endian.Uint16(p.getBytes(2)))
+	/* Comeback = */ p.getBytes(4)
```

Comment:
comeback?

Reply:
Unused field; that's the name that the kernel calls it. Expanded the comment a bit.